LOOP-1717: Fix settings navigation to not dismiss first

rickpasetto opened the pull request:
- Get rid of SettingsTableViewController (yay!)
- Use `show` instead of `present` to avoid the intermediate dismiss

(Note: because the contract for vended ViewControllers from plugins "implicitly" creates a UINavigationController, we can't use Swift UI navigation without major changes.)

[LOOP-1717](https://tidepool.atlassian.net/browse/LOOP-1717)

rickpasetto committed:
Remove SettingsTableViewController

rickpasetto committed:
Don't dismiss; Use `show` instead of `present`

rickpasetto committed:
Rename some stuff

nhamming reviewed (approved):
Just wondering if we are deleting `SettingsTableViewControler`, should all unused Views/VCS based on the settings swap be removed (e.g., `SettingsImageTableViewCell`, `ExplicitlyDismissibleModal`, etc.)?

Otherwise LGTM.

novalegra commented on Loop/Views/SettingsView.swift:
```diff
@@ -196,8 +189,6 @@ extension SettingsView {
             ActionSheet.Button.default(Text(availableService.localizedTitle)) {
-                // TODO: this "dismiss then call didTapAddService()" here is temporary, until we've completely gotten rid of SettingsTableViewController
-                self.dismiss()
                 self.viewModel.servicesViewModel.didTapAddService(availableService)
```
When I tap to add the service and successfully add one, the view doesn't refresh (I have to exit the settings screen and then re-open it for the changes to show up)

rickpasetto committed:
LOOP-1717: Navigation using sheets

Because of the current plugin vending strategy, we have to navigate using sheets.  Therefore, this makes the UI more consistent for all buttons, including Therapy Settings

rickpasetto committed:
Merge branch 'dev' into rick/swap-settings-wip3

rickpasetto committed:
More PR feedback: update service list when service is added/deleted

rickpasetto committed:
Merge branch 'dev' into rick/swap-settings-wip3

novalegra commented on Loop/Views/SettingsView.swift:
```diff
@@ -89,19 +90,22 @@ extension SettingsView {
-                                                    chartColors: .primary,
-                                                    didSave: viewModel.didSave))) {
-                LargeButton(action: { },
+            LargeButton(action: { self.showTherapySettings = true },
```
Want to check it's intentional that this screen is now presented modally

rickpasetto replied:
Yes.  This is very intentional ... based on Design feedback: they do not want some screens to show as a "left-right slide" navigation (TherapySettings) and others as a sheet (modal).  This makes them consistent.

novalegra commented on Loop/Views/SettingsView.swift:
```diff
@@ -89,19 +90,22 @@ extension SettingsView {
                             label: NSLocalizedString("Therapy Settings", comment: "Title text for button to Therapy Settings"),
                             descriptiveText: NSLocalizedString("Diabetes Treatment", comment: "Descriptive text for Therapy Settings"))
+                .sheet(isPresented: $showTherapySettings) {
+                    TherapySettingsView(
```
I'm not seeing a title now for therapy settings (though it shows up fine for the pump simulator and Dexcom screens)

rickpasetto replied:
Really?

![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-28 at 12 54 06](https://user-images.githubusercontent.com/308993/91610158-92910880-e92d-11ea-848a-0ea34e0cec12.png)

Shows fine for me...what am I missing?

novalegra replied:
Hmm, showing up fine now when I re-checked-out. Probably had an incorrect branch checked out (I'm betting LoopKit)

rickpasetto replied:
Yes `TherapySettingsView` is in `LoopKit[UI]`, so that would explain it 👍

novalegra reviewed (commented):
Noticed a few changes that I want to check were intentional :)

rickpasetto committed:
Merge branch 'dev' into rick/swap-settings-wip3

novalegra reviewed (approved):
LGTM!
